dev-notes/release-process.md

```diff
@@ -0,0 +1,161 @@
+# Instructions for releasing a new version
+
+Deprecates release instruction from
+[RFC 2](https://github.com/edgedb/rfcs/blob/master/text/0002-edgedb-release-process.rst).
+
+EdgeDB packages are published on https://packages.edgedb.com.
+They are build in GitHub Actions pipelines, using
+https://github.com/edgedb/edgedb-pkg.
+
+Releases are built from a release branch associated with a major version
+(i.e. "release/4.x"). At feature freeze, we create this branch. From that moment
+on, all additional commits will have to be cherry-picked to this branch.
+
+Before the major version, we publish "testing releases":
+
+- "alpha" (i.e. `v4.0a1`, `v4.0a2`),
+- "beta" (i.e. `v4.0b1`, `v4.0b2`),
+- "release candidates" (`v4.0rc1`) that we might promote into the final release.
+
+## Preparing commits for a release
+
+We have GitHub labels (i.e. "backport-4.x") associated with each major release
+that indicate PRs that should be backported to the release branch.
+
+It should be sufficient to go over that list and cherry-pick PR commits to the
+release branch.
+
+Sometimes, people will forget to tag the PR to be back-ported, so a good
+practice is to list all commits since the last release:
+
+```
+git show releases/4.x # to see the last commit that has been cherry-picked
+git log master # find the hash of that commit on master
+git log hash_of_that_commit..master > ../to-backport.txt
+```
+
+Now, one can go trough the list and see if the commits are worth back-porting.
+A few pointers:
```

Comment:
Generally I'd say don't backport features also, unless it's high priority in some way